### ISSUE TITLE
Exploring Visual Explanations for Contrastive Language-Image Pre-training

### ISSUE THREAD

**Issue by ooooohira** [ooooohira/paper_summary#12]
### 論文リンク
[https://arxiv.org/abs/2209.07046](https://arxiv.org/abs/2209.07046)

### 著者
Yi Li, Hualiang Wang, Yiqun Duan, Hang Xu, Xiaomeng Li

### 投稿年
Thu, 15 Sep 2022

## 新規性や利点
<p align="center">
<img width="503" alt="スクリーンショット 2024-04-25 23 40 39" src="https://github.com/ooooohira/paper_summary/assets/131585401/5cb63fbd-fb69-4273-9655-491f6c20470a">
</p>

誤差逆伝播なしに低コストでITSMで物体内部に注目させることが可能になった。

## 提案手法
<img width="1133" alt="スクリーンショット 2024-04-25 23 59 15" src="https://github.com/ooooohira/paper_summary/assets/131585401/3e97e3e1-67ba-4242-ba8d-565c39a03cbc">

分析によってmax poolingを用いることで、物体の外側でなく物体自身を見てくれるようになった。そこでこのことを活用して、ECLIPという説明可能性の高いCLIPを、CLIPに対して追加のアーキテクチャー(画像のみについてのattentionでうまく物体の内側を注視できている(しかし、ITSMではうまくいっていないがマシだと考えて)すなわちDINOなどのself-attention mask)を加えることで考案した。

また、ITSMで外側を見ていることをもとに1-(元のITSM)をITSMとするRCLIPという手法を編み出した。

この研究は、おそらくただ物体内部に注目しているITSMを得るための手法を考案しているだけ。

## 実験
mIoUとmSCという手法で、ITSMでうまく物体の中身を見れているか確認し、比較する。

## 結果と考察
<img width="1382" alt="スクリーンショット 2024-04-26 0 06 48" src="https://github.com/ooooohira/paper_summary/assets/131585401/d73e37b4-6288-4081-8b42-f16633cb129e">

## 感想や疑問点
ITSMの考え方は面白く、納得できた。

feature shiftはばらけていること、 semantic shiftはばらけている結果内側を見たり外側を見たりすることなのか？
(3),(4),(9)式は間違っているはずφi→φt。
しかも、(9)の右と左の式の次元合わなくないのか？？
ITSMの可視化の際にはNt方向にaverageをとっているのか。
learnable parameterのφiとφtは学習済みのはず。
Figure 2.の(b),(c)はどのように可視化しているのか。
DINOの詳細を知りたい。
attention poolingはチャンネル方向に重みつきの和をとることなのか？→Fi*φiの操作ってこと？
図4では、最終層の値を用いているため、各チャンネルでの位置は意味をなさないのではないのか？？
(6),(7)式が理解できない。DINOとかではsemantic shiftが起きていないから、それのattention maskを利用させてもらうというマインド？？
なぜITSM作る時FiをMasked Max Poolingに通さないのか？？
FmcのサイズはFiと同じか？？
